ArduPlane/geofence.cpp

```diff
@@ -280,6 +280,52 @@ bool Plane::geofence_check_maxalt(void)
     return (adjusted_altitude_cm() > (g.fence_maxalt*100.0f) + home.alt);
 }
 
+/*
+  pre-arm check for being inside the fence
+ */
+bool Plane::geofence_prearm_check(void)
+{
+    if (!geofence_enabled()) {
+        gcs().send_text(MAV_SEVERITY_WARNING, "PreArm: Fence not enabled");
+        return false;
+    }
+
+    /* allocate the geo-fence state if need be */
+    if (geofence_state == nullptr || !geofence_state->boundary_uptodate) {
+        geofence_load();
+        if (!geofence_enabled()) {
+            // may have been disabled by load
+            gcs().send_text(MAV_SEVERITY_WARNING, "PreArm: Fence load failed");
+            return false;
+        }
+    }
+
+    if (geofence_state->floor_enabled && g.fence_minalt != 0) {
+        // can't use minalt with prearm check
+        gcs().send_text(MAV_SEVERITY_WARNING, "PreArm: Fence floor enabled");
```

Comment:
This is unclear.  Burn the bytes for something longer :-)  "FENCE_AUTOENABLE=3 requires floor disabled"?